### PR TITLE
[#14] [Android] [UI] As a user, I can see a list of surveys on the home screen

### LINKED ISSUE #14: [Android] [UI] As a user, I can see a list of surveys on the home screen
## Why

After logging in, the user can see the Home screen with the survey list.

## Acceptance Criteria

- Show the horizontal list with the item detail as:
  - Show the indicator
  - Show the title and description
  - Show the arrow button

Design: https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=44%3A148&scaling=scale-down

image

### PR DESCRIPTION
**Pull request by luongvo**
- Close #14

## What happened 👀

- Implement Home Surveys UI with a horizontal survey list.
- Implement Home Footer UI with title, description & Next button.
- Implement horizontal dot indicator.

## Insight 📝

- Use [Accompanist Pager](https://google.github.io/accompanist/pager/) to build the horizontal survey list.
- Adjust `DimmedImageBackground` to show the background image from an URL.
- Use [Accompanist Pager Indicators](https://google.github.io/accompanist/pager/#indicators) to build the horizontal dot indicator for pager.

## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/209780035-064c489e-5e9d-48dd-8cb5-d5e8ab107787.mp4

### CODE REVIEW

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt [hunk 1]
```diff
@@ -20,11 +18,22 @@ fun HomeScreen(
...
 private fun HomeScreenContent() {
     Box(
-        modifier = Modifier
-            .fillMaxSize()
-            // TODO demo background, integrate in https://github.com/luongvo/kmm-survey/issues/16
-            .background(Color(0xFF8E9398))
+        modifier = Modifier.fillMaxSize()
     ) {
+        HomeSurveys(
+            listOf(
```
Comment (blyscuit) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt]: We might need `todo` here.

**Review by blyscuit — APPROVED** [review]

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt (hunk 1 above)
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt]: Fixed in https://github.com/luongvo/kmm-survey/pull/61/commits/c9a6d2a223230e4cb658f59ea145231bf6df66f8

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/NextCircleButton.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.ui.graphics.Color.Companion.White
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import vn.luongvo.kmm.survey.android.R
+import vn.luongvo.kmm.survey.android.ui.theme.AppTheme.dimensions
+
+@Composable
+fun NextCircleButton(
```
Comment (doannimble) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/NextCircleButton.kt:20]: @luongvo Not a big due, just optional. As it is a common button, what do you think if we name this like `CircleButton` with the drawable id parameter?
I understand that we do not reuse with other drawable so far. We can do the refactor later if need.

**Review by doannimble — APPROVED** [review]
Look good 👍

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/NextCircleButton.kt (hunk 2 above)
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/NextCircleButton.kt:20]: @doannimble Yep, valid point. I will consider refactoring if needed to expand it to be more common with a drawableId param later 👍

**Pull request merged**